server/client_rest.go

```diff
@@ -31,6 +31,19 @@ type clientRest struct {
 	httpClient http.Client
 }
 
+func (client *clientRest) RegisterIdentity(identity *dto_discovery.Identity) (err error) {
```

Comment:
We should reconsider our pointer convention.
Because now You can easily pass `nil` which unwanted situation

Reply:
1. Lets change signature of `IdentityManager.GetIdentity() *Identity` -> `IdentityManager.GetIdentity() Identity, error`
2. Same with all other `IdentityManager` methods, which returns pointer *Identity
3. Lets stop passing *Identity in all project which can easily become `nil`

Reply:
Identities should be immutable, indestructible and persisted on myst network.